src/game/puppyprint.c

```diff
@@ -1389,13 +1376,14 @@ void print_small_text_buffered(s32 x, s32 y, const char *str, u8 align, s32 amou
     sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 5] = gCurrEnvCol[1];
     sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 6] = gCurrEnvCol[2];
     sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 7] = gCurrEnvCol[3];
-    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 8] = strLen;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 8] = strLen + 1;
     sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 9] = align;
     sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 10] = (amount == -1) ? 255 : amount;
```

Comment:
I honestly don't remember whether `amount` has any implications with the `strlen` null termination (or rather cutting the string short before it reaches a null terminator), but I think it does if my memory serves me right. This is an annoying problem to solve because I think it doesn't account for if you have any commands in the string, which means you can't just use it to decrease the size of `strLen`, right? Honestly idk what to suggest here, apart from bugfixing the source wherever it happens.